### PR TITLE
Add initial set of guide documents.

### PR DESCRIPTION
**Pull request by cyharyanto**
### Initial SDLC Guide Setup

This PR establishes the foundation for our OKR-based Software Development Life Cycle (SDLC) guide. Key additions include:
- `README.md`: Overview of the repository and guide structure
- `CONTRIBUTING.md`: Guidelines for contributing to the guide
- `LICENSE`: MIT License for the project
- `docs/introduction.md`: Detailed introduction to our OKR-based SDLC approach
- `docs/timeframes-and-milestones.md`: Explanation of timeframes and milestones in our SDLC
- `docs/why-okr.md`: Deeper explanation about OKRs
- `docs/template-guide.md`: Guide for using the OKR template with LLM tools
- `templates/okr.md`: Template for creating OKRs using LLM tools

These documents provide a comprehensive starting point for team members to understand and implement our SDLC process, with a focus on OKRs at the Product, Team, and Individual levels.

### CODE REVIEW

File: docs/template-guide.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+1. Open Anthropic Console and start a new conversation.
+2. Paste the entire template into the input box.
+3. Replace the variables with your specific information. For example:
+
+```
+<name>ZipThought Engineering Team</name>
+<level>Team</level>
+<context>ZipThought is a SaaS company developing a collaborative project management tool. The engineering team is responsible for backend infrastructure and API development.</context>
+<related>Product Team OKR: Increase user engagement by 25% this quarter</related>
```
Comment (milmigi) [inline docs/template-guide.md:45]: for consistency purpose, this should be `related_okrs`?

File: docs/timeframes-and-milestones.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+- Deploy major system updates
+- Conduct end-of-quarter review
+- Evaluate OKR achievement at all levels across all objectives
+- Begin planning for next quarter
+
+Throughout this process, higher-level OKRs guide the creation of lower-level OKRs, ensuring alignment across the organization. Regular check-ins allow for adjustments, considering progress across multiple objectives and both frontend and backend aspects of development.
+
+## Adapting to Different Project Sizes
+
+- **Shorter Projects** (1-2 months): May focus on critical backend improvements or specific frontend features, aligning closely with immediate Product OKRs.
```
Comment (milmigi) [inline docs/timeframes-and-milestones.md:103]: In shorter projects, for example 1 month project. How's the OKR process would be like? I imagine spending 1 to 2 weeks is quite expensive.

File: docs/template-guide.md (hunk 1 above)
Comment (cyharyanto) [reply docs/template-guide.md:45]: Ah yes. we'll need to also update the template itself.

File: docs/timeframes-and-milestones.md (hunk 2 above)
Comment (cyharyanto) [reply docs/timeframes-and-milestones.md:103]: We can think the OKR setting as part of the planning process. Counterintuitively, the ideal time for planning is about 25% [1], well above the 15% reported by most people. But personally, we should spend at least 15% dedicated for planning with 25% as the upper limit. Hence, if we have a 1-month project, to have 1 week as planning is good while 2-week is definitely excessive (with some exceptions).

**References**
[1] P. Serrador and R. Turner, “What is Enough Planning? Results From a Global Quantitative Study,” IEEE Trans. Eng. Manage., vol. 62, no. 4, pp. 462–474, Nov. 2015, doi: [10.1109/TEM.2015.2448059](https://doi.org/10.1109/TEM.2015.2448059).

Comment (milmigi) [reply docs/timeframes-and-milestones.md:103]: Yeah, 1 week of planning sounds good for 1 month project. I was imagining between 3 days to 1 week.

**Review by milmigi — APPROVED** [review]

**Pull request merged**